lib/typescript/render/shared.ts

```diff
@@ -16,21 +15,28 @@ export interface SuggestedReplyCommand extends Command {
 
 export type CommandUnion = SuggestedReplyCommand;
 
-export interface MessageRenderProps {
-  message: RenderedContent;
+interface RenderProps {
+  contentType: 'message' | 'template';
+  renderedContent: RenderedContentUnion;
   source: string;
-  contact?: Contact;
-  lastInGroup?: boolean;
+}
+
+export interface MessageRenderProps extends RenderProps {
+  contentType: 'message';
   invertSides?: boolean;
   commandCallback?: (command: CommandUnion) => void;
 }
 
-export const getDefaultMessageRenderingProps = (props: MessageRenderProps): DefaultMessageRenderingProps => {
-  const fromContact = isFromContact(props.message) ?? null;
+export interface TemplateRenderProps extends RenderProps {
+  contentType: 'template';
+}
+
+export type RenderPropsUnion = MessageRenderProps | TemplateRenderProps;
+
+export const getDefaultRenderingProps = (props: RenderPropsUnion): DefaultRenderingProps => {
+  const fromContact = isFromContact(props.renderedContent);
+
   return {
     fromContact,
-    contact: props.contact ?? null,
-    commandCallback: props.commandCallback,
-    sentAt: props.lastInGroup ? formatTime(props.message.sentAt) : null,
   };
```

Comment:
All the logic exclusive to rendering messages is moved to the wrapper <InfoMessageWrapper />, which displays the Avatar and sentAt time along with the message's content. This wrapper is also responsible for determining is which side the messages are displayed.

-  _RenderProps__ contains the props common to all the content  that the library needs to render (i.e. messages and templates)
-  _DefaultRenderingProps_  are needed by some components that the library needs to render, across all content (this is needed for all the components that use <Text />, as this props will determine the background-color of the message)